File: app/src/main/java/com/apollopharmacy/vishwam/ui/home/SubmenuAdapter.java
package com.apollopharmacy.vishwam.ui.home;

import android.content.Context;
import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.apollopharmacy.vishwam.R;
import com.apollopharmacy.vishwam.databinding.SubmenuItemRowBinding;

import java.util.ArrayList;

public class SubmenuAdapter extends RecyclerView.Adapter<SubmenuAdapter.ViewHolder> {
    Context context;
    ArrayList<MenuModel> submenus;
    MainActivityCallback callback;

    public SubmenuAdapter(Context context, ArrayList<MenuModel> submenus, MainActivityCallback callback) {
        this.context = context;
        this.submenus = submenus;
        this.callback = callback;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        SubmenuItemRowBinding submenuItemRowBinding = DataBindingUtil.inflate(LayoutInflater.from(context), R.layout.submenu_item_row, parent, false);
        return new ViewHolder(submenuItemRowBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.submenuItemRowBinding.menuName.setText(submenus.get(position).getMenuName());
        holder.submenuItemRowBinding.menuIcon.setImageResource(submenus.get(position).getMenuIcon());
        holder.submenuItemRowBinding.menu.setOnClickListener(v -> {
            callback.onClickSubmenuItem(submenus.get(position).getMenuName(), submenus, position);
        });
    }

    @Override
    public int getItemCount() {
        return submenus.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        SubmenuItemRowBinding submenuItemRowBinding;

        public ViewHolder(@NonNull SubmenuItemRowBinding submenuItemRowBinding) {
            super(submenuItemRowBinding.getRoot());
            this.submenuItemRowBinding = submenuItemRowBinding;
        }
    }
}

File: app/src/main/java/com/apollopharmacy/vishwam/ui/home/discount/filter/adapter/SubMenuAdapter.kt
package com.apollopharmacy.vishwam.ui.home.discount.filter.adapter

import android.accounts.AccountManagerCallback
import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.apollopharmacy.vishwam.R
import com.apollopharmacy.vishwam.data.model.discount.FilterData
import com.apollopharmacy.vishwam.databinding.PendencySitesBinding
import com.apollopharmacy.vishwam.databinding.RtoPendencySitesBinding
import com.apollopharmacy.vishwam.databinding.RtoSitesBinding
import com.apollopharmacy.vishwam.databinding.SubmenuFilterAdapterBinding
import com.apollopharmacy.vishwam.ui.home.qcfail.model.Getqcfailpendinghistoryforhierarchy
import com.apollopharmacy.vishwam.ui.home.qcfail.model.PendingCountResponse
import com.apollopharmacy.vishwam.ui.home.qcfail.model.QcDashBoardCallback
import java.text.DecimalFormat
import java.text.NumberFormat
import java.util.*
import kotlin.collections.ArrayList

class SubMenuAdapter(
    val mContext: Context,

    var filterData: List<FilterData>,
) :
    RecyclerView.Adapter<SubMenuAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {

        val submenuFilterAdapterBinding: SubmenuFilterAdapterBinding =
            DataBindingUtil.inflate(
                LayoutInflater.from(mContext),
                R.layout.submenu_filter_adapter,
                parent,
                false
            )
        return ViewHolder(submenuFilterAdapterBinding)

    }

    @SuppressLint("ResourceAsColor")
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val items = filterData.get(position)
        var menuFilter = -1
        holder.submenuFilterAdapterBinding.menuTitle.setText(items.MenuTitle)


    }


    override fun getItemCount(): Int {
        return filterData.size
    }

    class ViewHolder(val submenuFilterAdapterBinding: SubmenuFilterAdapterBinding) :
        RecyclerView.ViewHolder(submenuFilterAdapterBinding.root)
}


